dataform/examples/dataform_datavault/README.md

```diff
@@ -0,0 +1,107 @@
+
+### Example DataVault implementation on BigQuery by using Dataform
+
+Use Case: As an energy company would like to store, analyze and provide insights on data from meters and meter readings. 
+We have stored source data in staging dataset about meter and meter readings. And, in "raw" dataset datavault tables(hubs, links & satellites) are created in BigQuery using dataform lib. 
+
+The sqlx files for hubs, satellites & links are placed under [definitions](https://github.com/prabhaarya/bigquery-utils/tree/feature/dataform-datavault/dataform/examples/dataform_datavault/definitions/raw) folder and any functions used needs to placed under [includes](https://github.com/prabhaarya/bigquery-utils/tree/feature/dataform-datavault/dataform/examples/dataform_datavault/includes) folder. 
+
+### Source table schemas:
+Meter Staging table:
+
+  CREATE OR REPLACE TABLE staging.meter_load
+  (
+  meter_id INTEGER,
+  manufacturer STRING,
+  service_name STRING,
+  source STRING,
+  loaddate DATE
+  );
+
+Meter Readings Staging table:
+
+  CREATE OR REPLACE TABLE staging.meter_readings_load
+  (
+  readings_id INTEGER,
+  meter_id INTEGER,
+  readings FLOAT64,
+  readings_time TIMESTAMP,
+  source STRING,
+  loaddate DATE
+  );
+
+### Source table exports:
+1. meter_load --> Meter table contains data about meter.
+   ![](screenshots/meter_load.png)
+
+2. meter_readings_load --> Meter readings table contains data about meter readings.
+   ![](screenshots/meter_readings_load.png)
+
+## How to use?
+
+### Pre-requsites:
+1. npm lib needs be installed
+2. [Google cloud project](https://developers.google.com/workspace/guides/create-project) should be available
+3. BigQuery api is enabled
+4. Access to create service account keys
+5. [Dataform lib ](https://docs.dataform.co/dataform-cli) lib using NPM
+
+
+### Steps to follow:
+All the steps below can be used via terminal or Use the CloudShell editor.
+
+In order to test and run the dataform, please follow below steps:
+
+1. Initialize a Dataform project and then create a new Dataform google cloud project ([*optional if you already have one*])
+
+- dataform init bigquery dataform-demo --default-database $(gcloud config
+  get-value project) --include-schedules
+
+2. Set up authentication to BigQuery
+    - Go to https://console.cloud.google.com/iam-admin/serviceaccounts and
+      create a service account. Give this account the role of BigQuery Admin (it helps Dataform can create new tables etc.). Then, download the JSON key to
+      the project and upload the file to CloudShell. (optional if you already have a account with service account key uploaded into *optional if you already have a account with service account key uploaded into it*)
+    - Using terminal:
```

Comment:
In  martkup you can add code block with ticks. Let's convert these to code blocks

Reply:
```shell
cd dataform-demo
dataform init-creds bigquery
# Provide the path to the JSON key file when asked. Then make sure to add the key file to .gitignore so that you don't check it in by mistake.
echo filename.json >> .gitignore
git add -f .gitignore
```